io/zenoh-transport/src/unicast/establishment/ext/auth/mod.rs

```diff
@@ -604,13 +607,17 @@ impl<'a> AcceptFsm for &'a AuthFsm<'a> {
             match (self.usrpwd.as_ref(), state.usrpwd.as_mut()) {
                 (Some(e), Some(s)) => {
                     let x = ztake!(exts, id::USRPWD);
-                    e.recv_open_syn((s, ztryinto!(x, S))).await?;
+                    let username = e.recv_open_syn((s, ztryinto!(x, S))).await?;
+                    let user_passwd_id = UsrPwdId(Some(username));
+                    return Ok(user_passwd_id);
+                }
+                (None, None) => {
+                    return Ok(UsrPwdId(None));
                 }
-                (None, None) => {}
                 _ => bail!("{S} Invalid UsrPwd configuration."),
             }
         }
-
+        #[cfg(not(feature = "auth_usrpwd"))]
```

Comment:
Always return the same type, conditionally enable/disable fields inside the type.